### PR TITLE
FSE: Update Homepage Posts block branding

### PR DESCRIPTION
**Pull request by marekhrabe**
#### Changes proposed in this Pull Request

Update the block name as per paYJgx-li-p2. To make the connection between visual side and code more clear, also update all related directory names.

#### Testing instructions

* Build FSE plugin based, based on its readme instructions
* Try this plugin on your WP installation
* In the UI, the block should be visible as "Blog Posts"
* When viewing the post source, it should show up as `a8c/blog-posts`
* Functionality should be untouched and block is expected to work both in editor and on the frontend

### CODE REVIEW

**Comment by retrofox** [comment]
Are we going to close https://github.com/Automattic/wp-calypso/tree/try/fse-nha-sync?

**Comment by marekhrabe** [comment]
> Are we going to close…

This is based off that branch and assumes both PRs will be merged eventually (in the order of their creation)

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php
Comment (obenland) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php]: 😢

Comment (retrofox) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php]: I asked about this as well. 😅

File: apps/full-site-editing/package.json
Comment (retrofox) [inline apps/full-site-editing/package.json]: why not instead of setting a name which could change again (🤞) define a name not related to the FSE plugin name? 

`sync:newspack-articles`, for instance. In the end, it's what it does.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php
Comment (retrofox) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php]: ⚠️ This path is wrong: `failed to open stream` :

```
[04-Dec-2019 18:45:40 UTC] PHP Warning:  require_once(/var/www/html/wp-content/plugins/full-site-editing-plugin/blog-posts/index.php): failed to open stream: No such file or directory in /var/www/html/wp-content/plugins/full-site-editing-plugin/full-site-editing-plugin.php on line 338
```

**Review by retrofox — DISMISSED** [review]
Going to `Request changes` to surface this [issue](https://github.com/Automattic/wp-calypso/pull/38171/files#r353919775).

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php
Comment (obenland) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php]: Resolved in 5138e90

File: apps/full-site-editing/package.json
Comment (obenland) [reply apps/full-site-editing/package.json]: Newspack-articles doesn't seem to be used anywhere else. I think `blog-post-block` makes sense, given that we're syncing that block.

**Review by obenland — APPROVED** [review]
I'm not a fan of the array syntax updates, but that shouldn't hold up this PR.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php
Comment (marekhrabe) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php]: I'm 99% sure this was done for me by the precommit hook. I haven't touched the syntax and I'd use `[]` on my own

**Comment by marekhrabe** [comment]
> I'm not a fan of the array syntax updates

Let's do a spring cleaning PR once we have all changes merged and before we deploy the block. I think those changes are done automatically by the pre-commit hook which brings question whether I (or others) have it correctly installed and configured. Loops back to #38011 and #34552

**Pull request merged**
